Add first working solution

ondrejsevcik opened the pull request:
TODO:

- [ ] CodeName as constant or attribute - don't know what's better for intended usage
- [ ] Create one executable file that could be easily distributed
- [ ] Improve README
- [ ] Add CI integration
- [ ] Improve logging - I've added very simple logging into the console, maybe that's enough ❔

petrsvihlik commented on KenticoCloudDotNetGenerators/CodeGenerator.cs:
```diff
@@ -0,0 +1,125 @@
+                    var property = Property.FromContentType(element.Codename, element.Type);
+                    classDefinition.AddProperty(property);
+                }
+                catch (InvalidOperationException e)
```
solve build warnings

petrsvihlik commented:
Thx Ondra, it works great and the code looks fine. Please check out my review notes + the following remarks:

ad your remarks:
- CodeName - let's solve it in the next iteration when we get to use the metadata (the same goes for url_slug)
- Create one executable - I'll set up appveyor after merging this PR
- Improve README - will do that after merging this PR
- Add CI integration - ditto
- Improve logging - not necessary, CLI logging is just enough

general feedback:
- any reason why you decided to go with .net 4.5 and not .net core?
  - afaik, the referenced libraries support .net standard...
  - the sdk supports .net standard so I guess we should make this generator run on other platforms as well (can do that later in a separate task)
  - we could use Microsoft's https://www.nuget.org/packages/Microsoft.Extensions.CommandLineUtils/

ondrejsevcik commented on KenticoCloudDotNetGenerators.Tests/CodeGeneratorTests.cs:
```diff
@@ -0,0 +1,27 @@
+    public class CodeGeneratorTests
+    {
+        [TestCase]
+        public void IntegrationTest()
```
We should add way to test DeliveryClient without making actual HTTP requests to the delivery API. That project I'm referencing in that test is my Sandbox project where I do all kinds of experiments, so I can't ensure that content type will stay there forever.

petrsvihlik replied:
right. would it make sense to point it to the same project we reference from the SDK? (or apiary)?

ondrejsevcik replied:
The SDK references my Sandbox project as well so there is the same issue 😄

petrsvihlik replied:
ok, see that now. part of the tests use your project. 

we should use this one: 975bf280-fd91-488c-994c-2f04416e5ee3

ondrejsevcik commented:
Good point, I didn't think about that. I will convert it to .NET core project.

ondrejsevcik commented:
I rewrote it to .NET Core. I hope it's what you've asked for :)
I also changed the command line parser to Microsoft's one because CommandLineParser wasn't compatible with .NET Core. 

If you find anything else, just let me know ;)

Thank you

petrsvihlik commented:
thanks @ondrejsevcik . I made a few changes regarding the project structure and running self-contained on various platforms
